File: dotnet/tests/Datastructure/Otterop/Datastructure/TestLinkedList.cs
namespace Otterop.Datastructure
{
    public class TestLinkedList : Otterop.Test.TestBase
    {
        [Otterop.Test.Test]
        public void Add()
        {
            Otterop.Lang.String a = Otterop.Lang.String.Wrap("a");
            Otterop.Lang.String b = Otterop.Lang.String.Wrap("b");
            Otterop.Lang.String c = Otterop.Lang.String.Wrap("c");
            Otterop.Lang.String d = Otterop.Lang.String.Wrap("d");
            Otterop.Lang.String e = Otterop.Lang.String.Wrap("e");
            Otterop.Lang.Array<Otterop.Lang.String> strings = Otterop.Lang.Array.NewArray(5, a);
            strings.Set(0, a);
            strings.Set(1, b);
            strings.Set(2, c);
            strings.Set(3, d);
            strings.Set(4, e);
            Otterop.Lang.Array<Otterop.Lang.String> expected = Otterop.Lang.Array.NewArray(5, a);
            expected.Set(0, e);
            expected.Set(1, d);
            expected.Set(2, a);
            expected.Set(3, b);
            expected.Set(4, c);
            LinkedList<Otterop.Lang.String> l = new LinkedList<Otterop.Lang.String>();
            l.AddLast(a);
            l.AddLast(b);
            l.AddLast(c);
            l.AddFirst(d);
            l.AddFirst(e);
            AssertTrue(l.Size() == 5, Otterop.Lang.String.Wrap("Size should be 5"));
            int i = 0;
            foreach (Otterop.Lang.String s in l)
            {
                AssertTrue(s.CompareTo(expected.Get(i)) == 0, Otterop.Lang.String.Wrap("Element mismatch"));
                i++;
            }

        }

    }

}

File: java/datastructure/src/test/java/otterop/datastructure/TestLinkedList.java
package otterop.datastructure;

import otterop.lang.Array;
import otterop.lang.String;
import otterop.test.Test;
import otterop.test.TestBase;

public class TestLinkedList extends TestBase {

    @Test
    public void add() {
        String a = String.wrap("a");
        String b = String.wrap("b");
        String c = String.wrap("c");
        String d = String.wrap("d");
        String e = String.wrap("e");

        Array<String> strings = Array.newArray(5, a);
        strings.set(0, a);
        strings.set(1, b);
        strings.set(2, c);
        strings.set(3, d);
        strings.set(4, e);
        Array<String> expected = Array.newArray(5, a);
        expected.set(0, e);
        expected.set(1, d);
        expected.set(2, a);
        expected.set(3, b);
        expected.set(4, c);

        LinkedList<String> l = new LinkedList<String>();
        l.addLast(a);
        l.addLast(b);
        l.addLast(c);
        l.addFirst(d);
        l.addFirst(e);
        this.assertTrue(l.size() == 5, String.wrap("Size should be 5"));
        int i = 0;
        for (String s : l) {
            this.assertTrue(s.compareTo(expected.get(i)) == 0,
                    String.wrap("Element mismatch"));
            i++;
        }
    }
}
